### PR TITLE
Config Name Format

### LINKED ISSUE #711: Config name format
Yeah you're right, this will be a problem because internal config uses camelCase and yaml config uses kebab-case, so this will cause trouble in the future too. I think it would be best to stick to one format for both configs - camelCase is natural in JS/TS so I think it should be also used in yaml config. I will make a new issue for this but for now modify `getNewContributorConfig` to read from kebab-case and convert to camelCase manually.

_Originally posted by @whilefoo in https://github.com/ubiquity/ubiquibot/pull/689#discussion_r1314242213_

I think we should change yaml name format to camelCase to avoid having to convert it from kebab-case to camelCase when we parse it. This will also allow to have one unified Typescript type.

On this note I'd like to also discuss the separation of the config into two configs. One config would be EnvConfig which parses env variables and another would be the RepoConfig which parses the yaml. Having one merged config was causing issues in #684 because I had to validate the config and if it's invalid I still needed env variables to create Supabase adapter and create logger to log the error.

### PR DESCRIPTION
**Pull request by Sadaf-A**
kebab case to camel case

Resolves #711

### CODE REVIEW

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]
1/ You should not change the name of ubiquibot-config.yml
2/ I can see no changes in config names. 
3/ Please make sure you mark the PR `ready for review` once done.

**Comment by Sadaf-A** [comment]
> 1/ You should not change the name of ubiquibot-config.yml 2/ I can see no changes in config names. 3/ Please make sure you mark the PR `ready for review` once done.

i changed `price-multiplier` to `priceMultiplier` and could you please elaborate a little on where are the config names that I am supposed to change

**Comment by EtherealGlow** [comment]
Shouldn't you change every config name

**Comment by Sadaf-A** [comment]
> Shouldn't you change every config name 

I'm sorry but not able to understand can you link the file?

**Comment by 0xcodercrane** [comment]
We should change every config name in ubiquibot-config.yml and codebase to read them in src/utils/private.ts

**Comment by EtherealGlow** [comment]
Maybe also in default config json file

**Comment by 0xcodercrane** [comment]
> Maybe also in default config json file

True

**Comment by 0xcodercrane** [comment]
bump @EtherealGlow cause its blocking other PR from the merge.

**Comment by Sadaf-A** [comment]
@0xcodercrane please have a look now

**Comment by 0xcodercrane** [comment]
In the recent merge, we've changed the default config filename from `*.json` to `*.ts`. would you get the changes into your branch? @Sadaf-A

**Comment by Sadaf-A** [comment]
> In the recent merge, we've changed the default config filename from `*.json` to `*.ts`. would you get the changes into your branch? @Sadaf-A 

sure, I'll get to it right now

File: src/configs/ubiquibot-config-default.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-  "register-wallet-with-verification": false,
-  "promotion-comment": "\n<h6>If you enjoy the DevPool experience, please follow <a href='https://github.com/ubiquity'>Ubiquity on GitHub</a> and star <a href='https://github.com/ubiquity/devpool-directory'>this repo</a> to show your support. It helps a lot!</h6>",
-  "default-labels": [],
-  "time-labels": [
+  "evmNetworkId": 100,
+  "priceMultiplier": 1,
+  "issueCreatorMultiplier": 2,
+  "paymentPermitMaxPrice": 9007199254740991,
+  "maxConcurrentAssigns": 9007199254740991,
+  "assistivePricing": false,
```
Comment (whilefoo) [inline src/configs/ubiquibot-config-default.ts:9]: you don't need quotes (`"`) when using camelCase
```suggestion
  assistivePricing: false,
```
update this everywhere

**Review by whilefoo — CHANGES_REQUESTED** [review]

File: src/types/config.ts [hunk 2]
```diff
@@ -139,23 +139,23 @@ export type BotConfig = Static<typeof BotConfigSchema>;
...
-    "priority-labels": Type.Optional(Type.Array(LabelItemSchema)),
-    "payment-permit-max-price": Type.Optional(Type.Number()),
-    "command-settings": Type.Optional(Type.Array(CommandItemSchema)),
-    "promotion-comment": Type.Optional(Type.String()),
-    "disable-analytics": Type.Optional(Type.Boolean()),
-    "comment-incentives": Type.Optional(Type.Boolean()),
-    "assistive-pricing": Type.Optional(Type.Boolean()),
-    "max-concurrent-assigns": Type.Optional(Type.Number()),
+    "evmNetworkId": Type.Optional(Type.Number()),
+    "priceMultiplier": Type.Optional(Type.Number()),
```
Comment (whilefoo) [inline src/types/config.ts:143]:
```suggestion
    priceMultiplier: Type.Optional(Type.Number()),
```
it seems you didn't do this everywhere

File: src/types/config.ts [hunk 3]
```diff
@@ -173,24 +173,24 @@ export const WideOrgConfigSchema = Type.Composite([Type.Object({ "private-key-en
...
-  "disable-analytics": Type.Boolean(),
-  "comment-incentives": Type.Boolean(),
-  "assistive-pricing": Type.Boolean(),
-  "max-concurrent-assigns": Type.Number(),
+  "evmNetworkId": Type.Number(),
+  "priceMultiplier": Type.Number(),
+  "privateKeyEncrypted": Type.Optional(Type.String()),
+  "issueCreatorMultiplier": Type.Number(),
+  "timeLabels": Type.Array(LabelItemSchema),
+  "priorityLabels": Type.Array(LabelItemSchema),
```
Comment (whilefoo) [inline src/types/config.ts:181]:
```suggestion
  priorityLabels: Type.Array(LabelItemSchema),
```

File: src/utils/private.ts [hunk 4]
```diff
@@ -133,24 +133,24 @@ export const getWideConfig = async (context: Context) => {
...
-    priorityLabels: mergedConfigData["priority-labels"],
-    paymentPermitMaxPrice: mergedConfigData["payment-permit-max-price"],
-    disableAnalytics: mergedConfigData["disable-analytics"],
-    bountyHunterMax: mergedConfigData["max-concurrent-assigns"],
-    incentiveMode: mergedConfigData["comment-incentives"],
+    assistivePricing: mergedConfigData["assistivePricing"],
+    commandSettings: mergedConfigData["commandSettings"],
+    baseMultiplier: mergedConfigData["priceMultiplier"],
+    issueCreatorMultiplier: mergedConfigData["issueCreatorMultiplier"],
+    timeLabels: mergedConfigData["timeLabels"],
```
Comment (whilefoo) [inline src/utils/private.ts:142]:
```suggestion
    timeLabels: mergedConfigData.timeLabels,
```

**Review by whilefoo — CHANGES_REQUESTED** [review]

File: src/types/config.ts (hunk 2 above)
Comment (Sadaf-A) [reply src/types/config.ts:143]: I apologize for the oversight, and I appreciate your review. I will make the necessary adjustments immediately.

**Review by whilefoo — APPROVED** [review]
@0xcodercrane this might break the bot because it fetches the config from development branch but production bot won't have camelCase yet

**Comment by 0xcodercrane** [comment]
same feeling yeah. so my idea is to create a sync PR from development to main as soon as this PR merged.

**Comment by 0xcodercrane** [comment]
I am gonna handle the merge and hotfix if needed tomorrow morning.

**Comment by rndquu** [comment]
> @rndquu would you open the config PRs in both ubiquity and ubiquibot org for the upcoming changes?

I remember pavlovcik explicitly [stated](https://github.com/ubiquity/ubiquibot/pull/282#discussion_r1184663864) that the config should be kebab-case hence this [refactoring](https://github.com/ubiquity/ubiquibot/issues/299) was implemented. You 100% sure that we should migrate to camel case?

From my point of view it looks like that every 6 months we refactor the bot's config from camel case to kebab case and back which doesn't bring any advantages.

**Comment by pavlovcik** [comment]
> > @rndquu would you open the config PRs in both ubiquity and ubiquibot org for the upcoming changes?
> 
> I remember pavlovcik explicitly [stated](https://github.com/ubiquity/ubiquibot/pull/282#discussion_r1184663864) that the config should be kebab-case hence this [refactoring](https://github.com/ubiquity/ubiquibot/issues/299) was implemented. You 100% sure that we should migrate to camel case?
> 
> From my point of view it looks like that every 6 months we refactor the bot's config from camel case to kebab case and back which doesn't bring any advantages.

I vaguely recall a discussion we had where it was decided to finally change everything into camelCase recently because the only software that consumes this configuration file is the UbiquiBot typescript code, which expresses the same variables as camelCase.

**Review by pavlovcik — CHANGES_REQUESTED** [review]
CI doesn’t pass but the rest seem fine.

**Comment by Sadaf-A** [comment]
> CI doesn’t pass but the rest seem fine.

I'll fix that right away!

**Comment by 0xcodercrane** [comment]
> CI doesn’t pass but the rest seem fine.

ok will handle the merge after the conflicts resolved

**Comment by 0xcodercrane** [comment]
https://github.com/ubiquibot/ubiquibot-config/pull/8
https://github.com/ubiquity/ubiquibot-config/pull/15

**Comment by 0xcodercrane** [comment]
merging the both config PRs.

**Comment by 0xcodercrane** [comment]
https://github.com/ubiquity/ubiquibot/pull/795

sync PR

**Comment by 0xcodercrane** [comment]
> #795
> 
> sync PR

merged

**Comment by 0xcodercrane** [comment]
I had quick QAs once I merged the relevant PRs. Here are the QA issues.

Staging: https://github.com/ubiquibot/staging/issues/179
Production: https://github.com/ubiquibot/production/issues/65


So far looks good to me.

**Comment by EtherealGlow** [comment]
Finally

**Pull request merged**